source/common/network/connection_impl.cc

```diff
@@ -531,9 +531,9 @@ bool ConnectionImpl::bothSidesHalfClosed() {
 
 ClientConnectionImpl::ClientConnectionImpl(
     Event::Dispatcher& dispatcher, const Address::InstanceConstSharedPtr& remote_address,
-    const Network::Address::InstanceConstSharedPtr& source_address,
+    const Network::Address::InstanceRangeConstSharedPtr& source_address_range,
     Network::TransportSocketPtr&& transport_socket,
-    const Network::ConnectionSocket::OptionsSharedPtr& options)
+    const Network::ConnectionSocket::OptionsSharedPtr& options, Runtime::RandomGenerator& random)
```

Comment:
to your question I don't see this as an unreasonable amount of plumbing.  Often it's helpful to reviewers to squash things head to have one commit which is "plumb this unused thing through a bunch of APIs"  and a second which is "now use this thing".  Sometimes I even send them out separately.  But as Envoy plumbing changes go this doesn't seem egregious to me (opinions may vary =P) and it's not worth reorganizing after the fact.

The one thing that comes to mind to me, is that all this plumbing is (I think?) getting the ThreadLocalClusterManagerImpl's random number generator over to the dispatcher for local calls.  I wonder how it would look to instead change the ownership model and have the per thread dispatcher own the RNG, as I believe the ThreadLocalClusterManagerImpl already has ad dispatcher reference, and to my mind the thing which is already doing all your driving of events and scheduling having a random number generator for more nuanced driving-of-execution-path doesn't seem an unreasonable API addition.

Reply:
@htuch for a sanity check on this one before I suggest @curiouserrandy actually make any changes

Reply:
Oh, I'd missed that they were all references to a single rng.  blergh.  I still think it'd be handy to get it through the existing dispatcher reference but I'm less convinced that plumbing an rng through the "create dispatcher" function is going to be any cleaner so it's probably fine as is

Reply:
(Ooops, right, you mentioned the "plumb in one commit then use in a second"; I would have done that if I had remembered.)

Did you have any opinion on the other alternative plumbing path I raised, which is to put the RNG reference on the InstanceRange created and associated with the cluster, which means that it doesn't need to be plumbed through this path at all?  I haven't tried it yet so I'm not sure whether it's better, but it sorta makes sense to me--the InstanceRange exports bind(), and the RNG is necessary to implement bind(), so keeping the reference on the InstanceRange makes some sense to me.